ingestion_server/test/integration_test.py

```diff
@@ -66,575 +65,650 @@ def handle_task_callback():
-        assert self.__class__.cb_queue.get(timeout=120) == "CALLBACK!"
-
-    @pytest.mark.order(after="test_update_index_images")
-    def test_update_index_audio(self):
-        """Check that the audio data can be updated from the API database into ES."""
-
-        req = {
-            "model": "audio",
-            "action": "UPDATE_INDEX",
-            "since_date": "1999-01-01",
-            "index_suffix": "integration",
-            "callback_url": bottle_url,
-        }
-        res = requests.post(f"{ingestion_server}/task", json=req)
-        stat_msg = "The job should launch successfully and return 202 ACCEPTED."
-        self.assertEqual(res.status_code, 202, msg=stat_msg)
-
-        # Wait for the task to send us a callback.
-        assert self.__class__.cb_queue.get(timeout=120) == "CALLBACK!"
-
-    @pytest.mark.order(after="test_update_index_audio")
-    def test_index_deletion_succeeds(self):
-        self._ingest_upstream("audio", "temporary")
-        self._delete_index("audio", "temporary")
-
-    @pytest.mark.order(after="test_index_deletion_succeeds")
-    def test_alias_force_deletion_succeeds(self):
-        self._ingest_upstream("audio", "temporary")
-        self._promote("audio", "temporary", "audio-temp")
-        self._delete_index("audio", "temporary", "audio-temp")
-
-    @pytest.mark.order(after="test_alias_force_deletion_succeeds")
-    def test_alias_soft_deletion_fails(self):
-        self._ingest_upstream("audio", "temporary")
-        self._promote("audio", "temporary", "audio-temp")
-        self._soft_delete_index("audio", "audio-temp", "temporary")
-
-    @pytest.mark.order(after="test_alias_soft_deletion_fails")
-    def test_alias_ambiguous_deletion_fails(self):
-        # No need to ingest or promote, index and alias exist
-        self._soft_delete_index("audio", "audio-temp", "temporary", True)
-
-    @pytest.mark.order(after="test_alias_ambiguous_deletion_fails")
-    def test_stat_endpoint_for_index(self):
-        res = requests.get(f"{ingestion_server}/stat/audio-integration")
-        data = res.json()
-        assert data["exists"]
-        assert data["alt_names"] == ["audio-main"]
-
-    @pytest.mark.order(after="test_stat_endpoint_for_index")
-    def test_stat_endpoint_for_alias(self):
-        res = requests.get(f"{ingestion_server}/stat/audio-main")
-        data = res.json()
-        assert data["exists"]
-        assert data["alt_names"] == "audio-integration"
-
-    @pytest.mark.order(after="test_stat_endpoint_for_alias")
-    def test_stat_endpoint_for_non_existent(self):
-        res = requests.get(f"{ingestion_server}/stat/non-existent")
-        data = res.json()
-        assert not data["exists"]
+            }
+            res = es.search(index=index, query=query)
+            count = res["hits"]["total"]["value"]
+            assert (
+                count == 0
+            ), f"There should be no results that exactly match {sensitive_term} in {index}. Found {count}."
```

Comment:
I think we only used subtests for this because unittest doesn't support `pytest.mark.parametrize`. If it is possible to convert this to use parametrize and avoid including a new pytest plugin, that would be preferrable. I can't leave a normal PR review suggestion because the diff of this function is split on a weird diff match of `}` :confused:, but here's what I'm thinking might work to avoid the new dependency:

```py
@pytest.mark.order(
    after=["test_point_filtered_audio_alias", "test_point_filtered_image_alias"]
)
@pytest.mark.parametrize(
    "sensitive_term, index",
    zip(mock_sensitive_terms, ["audio-filtered", "image-filtered"]),
)
def test_filtered_indexes(sensitive_term, index):
    """
    Check that the sensitive terms are correctly filtered out.

    Each sensitive term should have zero exact matches for it when querying
    the filtered index. We use `term` against the `.raw` fields to ensure this.
    """
    es = _get_es()
    queryable_fields = ["title", "description", "tags.name"]
    query = {
        "bool": {
            "should": [
                {"term": {f"{field}.raw": sensitive_term}}
                for field in queryable_fields
            ]
        }
    }
    res = es.search(index=index, query=query)
    count = res["hits"]["total"]["value"]
    assert (
        count == 0
    ), f"There should be no results that exactly match {sensitive_term} in {index}. Found {count}."
```

Reply:
Thank you for the suggestion. This is a great to use parametrize instead of a third party plugin.